web/users/views.py

```diff
@@ -109,3 +116,79 @@ def search(self, request):
             ]
 
         return Response(users_results)
+
+
+class ConfirmClaimProfileView(APIView):
+    @method_decorator(never_cache)
+    def get(self, request, email, key):
+        # If record exists set is_claimed to true and assign user to placename
+        try:
+            record = ProfileClaimRecord.objects.get(profile_email=email)
+
+            with transaction.atomic():
+                print(record.is_claimed)
+                if not record.is_claimed:
+                    # Update Profile's owner
+                    user = User.objects.get(email=record.user_email)
+
+                    profile = record.profile
+                    profile.owner = user
+                    profile.save()
+
+
+                    # Update and Save record
+                    record.is_claimed = True
+                    record.save()
+
+                    return Response({
+                        'success': True,
+                        'message': 'You have successfully claimed your profile!',
+                        'profile': '{}/art/{}'.format(settings.HOST, _format_fpcc(record.profile.name))
+                    })
+                else:
+                    return Response({
+                        'success': False,
+                        'message': 'This profile was already claimed.',
+                    })
+        except ProfileClaimRecord.DoesNotExist:
+            return Response({
+                'success': False,
+                'message': 'This profile claiming link is invalid.'
+            })
+        except Exception as e:
```

Comment:
Please just let exceptions resolve rather than catching them. We want to be emailed in the case of an unknown exception.